screens/ChooseLocationScreen.tsx

```diff
@@ -0,0 +1,274 @@
+
+  const searchForAddress = async (address: string): Promise<void> => {
+    try {
+      const [{ latitude, longitude }] = await LocationProvider.geocodeAsync(
+        address
+      );
+
+      animateMapToRegion({ latitude: latitude, longitude: longitude });
+    } catch (e) {
+      alert('Address not found');
+    }
+  };
+
+  const insets = useSafeAreaInsets();
+  const theme = useTheme();
+  const styles = getStyles(theme, insets);
+
+  const animateMapToRegion = (_location: Location) => {
+    mapRef.current?.animateToRegion({
+      latitude: _location.latitude,
+      longitude: _location.longitude,
+      latitudeDelta: 0,
+      longitudeDelta: 0,
+    });
+  };
+
+  useEffect(() => {
+    setSelectedLocation(route.params.location);
+    if (route.params.location) {
+      animateMapToRegion(route.params.location);
+    }
+  }, [route.params.location]);
+
+  useEffect(() => {
+    const load = async () => {
+      setIsLoading(true);
+      const { success, hotspots = [] } = await loadHotspots();
+      setIsLoading(false);
+      if (!success) alert('Failed to load hotspots');
+      setHotspots(hotspots);
+    };
+    load();
+  }, []);
+
+  return (
+    <>
+      <SecondaryAppbar onBackPressed={() => navigation.goBack()} />
+      <View style={styles.mapContainer}>
+        <MapView
+          ref={mapRef}
+          // use intial region + animateToRegion instead of region as react state because
+          // gestures don't work well https://github.com/react-native-maps/react-native-maps/issues/3639
+          initialRegion={intialRegion}
+          showsUserLocation
+          style={styles.map}
+          onPress={async e => {
+            const { latitude, longitude } = e.nativeEvent.coordinate;
+            const isSameMarker =
+              selectedLocation?.longitude === longitude &&
+              selectedLocation?.latitude === latitude;
+
+            if (!isSameMarker) {
+              const location = await findPlace(latitude, longitude);
+              setSelectedLocation(location);
+            }
+          }}
+        >
+          {hotspots.map(h => (
+            <Marker
+              key={`${h.latitude} ${h.longitude}`}
+              coordinate={{
+                latitude: h.latitude,
+                longitude: h.longitude,
+              }}
+              onPress={() => navigation.navigate('Modal')}
+            >
+              <Image
+                source={getHotspotMarker(h)}
+                style={styles.marker}
+                resizeMode="contain"
+              />
+            </Marker>
+          ))}
+
+          {selectedLocation ? (
+            <Marker
+              key={`${selectedLocation.latitude} ${selectedLocation.longitude}`}
+              coordinate={{
+                latitude: selectedLocation.latitude,
+                longitude: selectedLocation.longitude,
+              }}
+              onPress={() => {
+                setSelectedLocation(defaultLocation);
```

Comment:
having a default location with coordinates 0, 0 doesn't seem a good idea. Maybe we should use a nullable here for `selectedLocation` and just reset it.

Reply:
done.